test/issues.jl

```diff
@@ -155,6 +155,32 @@ try
         end
     end
 
+    if AWS.DEFAULT_BACKEND[] isa AWS.HTTPBackend
+        @testset "issue 515" begin
+            # https://github.com/JuliaCloud/AWS.jl/issues/515
+            data = rand(UInt8, 100)
+            ATTEMPT_NUM = 0 # reset
+            patch = @patch function HTTP.request(args...; response_stream, kwargs...)
+                @show ATTEMPT_NUM += 1
+                if ATTEMPT_NUM == 1
+                    write(response_stream, rand(UInt8, 34)) # an incomplete stream that shouldn't be retained
+                    throw(HTTP.IOError(EOFError(), "msg"))
+                else
+                    write(response_stream, data)
+                end
+                headers = [
+                    "Content-Length" => string(length(data)),
+                ]
+                return HTTP.Response(200, "{\"Location\": \"us-east-1\"}")  # Made up region
```

Comment:
Dead code/old comment
```suggestion
                return HTTP.Response(200, "{\"Location\": \"us-east-1\"}")
```